src/land/smap_ssm2ioda.py

```diff
@@ -106,9 +105,12 @@ def _read(self):
             times = times[mask]
 
         # get datetime from filename
-        str_split = self.filename.split("_")
-        str_datetime = str_split[7]
-        my_datetime = datetime.strptime(str_datetime, "%Y%m%dT%H%M%S")
+        # file provides yyyy-mm-dd as an attribute
+        str_datetime = ncd.groups['Metadata'].groups['DatasetIdentification'].getncattr('creationDate')
+        my_datetime = datetime.strptime(str_datetime, "%Y-%m-%d")
```

Comment:
@YoulongXia-NOAA the file itself contains an attribute called `creationDate` this is just to the nearest day 

what is the update frequency of this product?